pkg/component/observability/monitoring/kubestatemetrics/resources.go

```
@@ -736,3 +712,14 @@ func (k *kubeStateMetrics) nameSuffix() string {
 	}
 	return suffix + k.values.NameSuffix
 }
+
+func (k *kubeStateMetrics) customResourceStateConfigMap() *corev1.ConfigMap {
+	customResourceStateConfig, err := yaml.Marshal(NewCustomResourceStateConfig())
+	utilruntime.Must(err)
```

We don't want to panic, please return the error instead.

---

Yes, you're right. I didn't return the error in the first place because the configuration that could error here is statically generated. But it definitely feels better to return the error and be on the safe-side 👍🏻